### PR TITLE
Properly parse empty priority in the UI

### LINKED ISSUE #156: TodoEditor requires setting a non-empty priority when saving
It should default to `none`

### PR DESCRIPTION
**Pull request by hobarrera**
Fixes #156

### CODE REVIEW

**Comment by hobarrera** [comment]
This contains the first *real* UI test. :tada:

**Comment by hobarrera** [comment]
Moving forward, I think it makes sense to have a `test_formatter` fixture which just returns `TodoFormatter` with the `DATE_FORMAT` and `TIME_FORMAT` we're using for tests.

**Comment by hobarrera** [comment]
@untitaker I hope you don't take reviewing this as an imposition, I just feel you'd like to review this (mostly because it sets a new example for tests) and you might have some comments or suggestions (or not).

File: tests/conftest.py [hunk 1]
```diff
@@ -69,6 +69,16 @@ def inner(tz='CET'):
     return inner
 
 
+@pytest.fixture
+def blank_todo(default_database):
+    todo = model.FileTodo(new=True)
```
Comment (untitaker) [inline tests/conftest.py]: `FileTodo`s are `new` by default.

File: tests/conftest.py [hunk 2]
```diff
@@ -69,6 +69,16 @@ def inner(tz='CET'):
     return inner
 
 
+@pytest.fixture
+def blank_todo(default_database):
```
Comment (untitaker) [inline tests/conftest.py]: I think it's more generally usable if this fixture returns a function that creates todos (eventually with a parameter for specifying list)

**Comment by hobarrera** [comment]
> Why make a difference between None and 0? They have the same semantics.

Basically one is "don't store anything in the file", and the other "store 0".  
I think this is a bit important, because if we edit a file, I'd like to avoid adding `PRIORITY:0\n` to it if the user didn't touch that field.

**Comment by hobarrera** [comment]
> This reintroduces the bug fixed by #160 because now we might again end up using humanized datetimes when prefilling TodoEditor.

Are you sure this comment was meant for this PR?

**Comment by untitaker** [comment]
>Are you sure this comment was meant for this PR?

That was a mistake I think

**Comment by untitaker** [comment]
It's up to you whether you want to make the model more complex for a nicer file diff.

**Pull request merged**